pkg/frontend/middleware/validate.go

```diff
@@ -23,100 +22,103 @@ const (
 
 var rxResourceGroupName = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
 
-func Validate(env env.Core, apis map[string]*api.Version) func(http.Handler) http.Handler {
-	return func(h http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			vars := mux.Vars(r)
-			route := mux.CurrentRoute(r)
+type ValidateMiddleware struct {
+	Location string
+	Apis     map[string]*api.Version
+}
 
-			apiVersion := r.URL.Query().Get(api.APIVersionKey)
+func (v ValidateMiddleware) Validate(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		route := mux.CurrentRoute(r)
 
-			if route == nil {
-				if log, ok := r.Context().Value(ContextKeyLog).(*logrus.Entry); ok {
-					log.Error("route was nil")
-				}
-				api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
-				return
-			}
+		apiVersion := r.URL.Query().Get(api.APIVersionKey)
 
-			if r.URL.Path != strings.ToLower(r.URL.Path) {
-				if log, ok := r.Context().Value(ContextKeyLog).(*logrus.Entry); ok {
-					log.Error("path was not lower case")
-				}
-				api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
-				return
+		if route == nil {
+			if log, ok := r.Context().Value(ContextKeyLog).(*logrus.Entry); ok {
+				log.Error("route was nil")
 			}
+			api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+			return
+		}
 
-			if _, found := vars["subscriptionId"]; found {
-				valid := uuid.IsValid(vars["subscriptionId"])
-				if !valid {
-					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
-					return
-				}
+		if r.URL.Path != strings.ToLower(r.URL.Path) {
+			if log, ok := r.Context().Value(ContextKeyLog).(*logrus.Entry); ok {
+				log.Error("path was not lower case")
 			}
-
-			if _, found := vars["resourceGroupName"]; found {
-				if !rxResourceGroupName.MatchString(vars["resourceGroupName"]) {
-					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceGroupNotFound, "", "Resource group '%s' could not be found.", vars["resourceGroupName"])
-					return
-				}
+			api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+			return
+		}
+
+		if _, found := vars["subscriptionId"]; found {
```

Comment:
Vars is going to go away (next ish step in my plan to remove gorilla), so I am not 100% sure it is worth doing it now since I will do it next anyways

Reply:
ack, fine to leave it for now